distributed/diagnostics/plugin.py

```diff
@@ -452,20 +455,29 @@ async def start(self, scheduler: Scheduler) -> None:
 
             if self.restart_workers:
                 nanny_plugin = _InstallNannyPlugin(self._install_fn, self.name)
-                await scheduler.register_nanny_plugin(
+                responses = await scheduler.register_nanny_plugin(
                     comm=None,
                     plugin=dumps(nanny_plugin),
                     name=self.name,
                     idempotent=True,
                 )
             else:
                 worker_plugin = _InstallWorkerPlugin(self._install_fn, self.name)
-                await scheduler.register_worker_plugin(
+                responses = await scheduler.register_worker_plugin(
                     comm=None,
                     plugin=dumps(worker_plugin),
                     name=self.name,
                     idempotent=True,
                 )
+            for response in responses.values():
+                if response["status"] == "error":
+                    response = {  # type: ignore[unreachable]
+                        k: deserialize(v.header, v.frames)
+                        for k, v in response.items()
+                        if isinstance(v, Serialized)
+                    }
```

Comment:
This is nasty, am I missing a better way? cc @fjetter

Reply:
I don't really know this code section very well but my first gut reaction is: No, this is it

Reply:
I suspect this is an artifact of https://github.com/dask/distributed/blob/f5c30e852b32acd2834f2cf5319bc8cab8556e6a/distributed/client.py#L5138C20-L5138C32

this explicit dumps shouldn't be necessary and I suppose this is why you get some `Serialized` objects here but IDK for sure